src/backend/commands/tablecmds.c

```diff
@@ -632,9 +665,20 @@ DefineRelation_int(CreateStmt *stmt,
 	 */
 	descriptor = BuildDescForRelation(schema);
 
-	localHasOids = interpretOidsOption(stmt->options);
+	localHasOids = interpretOidsOption(stmt->base.options);
 	descriptor->tdhasoid = (localHasOids || parentOidCount > 0);
 
+	/*
+	 * Check supported data types for pluggable format, i.e., orc
+	 * Need to remove this check if all data types are supported for orc format.
+	 */
+	if ((relkind == RELKIND_RELATION) && (relstorage == RELSTORAGE_EXTERNAL) &&
+	    formattername && pg_strncasecmp(formattername, "text", strlen("text")) &&
```

Comment:
TEXT/CSV format in pluggable storage should have no data type constraint, should remove this.